datafusion/physical-plan/src/joins/symmetric_hash_join.rs

```diff
@@ -587,6 +590,8 @@ struct SymmetricHashJoinStream {
     reservation: SharedMemoryReservation,
     /// Flag indicating whether there is nothing to process anymore
     final_result: bool,
+    /// Stream state for compatibility with HashJoinExec
+    state: HashJoinStreamState,
```

Comment:
I did not understand how this is used in SymmetricHashJoin. Can you explain it a bit further?